src/Groups/matrices/MatGrp.jl

```diff
@@ -184,29 +185,23 @@ function assign_from_description(G::MatrixGroup)
          @assert length(L) == 1
          G.X=L[1]
       end
-   elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1,G.deg,codomain(G.ring_iso))
-   elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1,G.deg,codomain(G.ring_iso))
-   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(-1,G.deg,codomain(G.ring_iso)))[1]
-   # TODO : we define explicitly orthogonal groups in dimension 1, since GAP does not support them
-   # GO(1,q) = { +1, -1 } and SO(1,q)=Omega(1,q) = { +1 }.
-   elseif G.descr==:GO
-      if G.deg==1
-         G.X=GAP.Globals.Group(GapObj([GapObj([-GAP.Globals.One(codomain(G.ring_iso))])]))
-      else
-         G.X=GAP.Globals.GO(0,G.deg,codomain(G.ring_iso))
-      end
-   elseif G.descr==:SO
-      if G.deg==1
-         G.X=GAP.Globals.Group(GapObj([GapObj([GAP.Globals.One(codomain(G.ring_iso))])]))
-      else
-         G.X=GAP.Globals.SO(0,G.deg,codomain(G.ring_iso))
-      end
+   elseif G.descr==Symbol("GO-") G.X=GAP.Globals.GO(-1, G.deg, F)
+   elseif G.descr==Symbol("SO-") G.X=GAP.Globals.SO(-1, G.deg, F)
+   elseif G.descr==Symbol("Omega-") G.X=GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(-1, G.deg, F))[1]
+   elseif G.descr==:GO G.X=GAP.Globals.GO(0, G.deg, F)
+   elseif G.descr==:SO G.X=GAP.Globals.SO(0, G.deg, F)
    elseif G.descr==:Omega
-      if G.deg==1
-         G.X=GAP.Globals.Group(GapObj([GapObj([GAP.Globals.One(codomain(G.ring_iso))])]))
-      else
-         G.X=GAP.Globals.SubgroupsOfIndexTwo(GAP.Globals.SO(0,G.deg,codomain(G.ring_iso)))[1]
-      end
+     # For even q or d = 1, \Omega(d,q) is equal to SO(d,q).
+     # Otherwise, \Omega(d,q) has index 2 in SO(d,q).
+     # Here d is odd, and we do not get here if d == 1 holds.
+     @assert G.deg > 1
```

Comment:
The function `assign_from_description` is called when one tries to access `G.X` for the groups in question.
For `omega_group` (i.e., `G.descr == Omega`), the case of dimension 1 is delegated to `SO`.
(Formally, one could create a `MatrixGroup` object `G`, then set `G.deg = 1` and `G.descr = Omega`, and then call `assign_from_description`. In this situation, one would indeed reach the `@assert` statement with `G.deg == 1`, but I think this would be asking or trouble.)